daemon/snap.go

```diff
@@ -99,6 +99,9 @@ func allLocalSnapInfos(st *state.State) ([]aboutSnap, error) {
 	for name, snapState := range snapStates {
 		info, err := snap.ReadInfo(name, snapState.CurrentSideInfo())
 		if err != nil {
+			if _, ok := err.(*snap.SnapVanishedError); ok {
```

Comment:
there are helpers in snapstate I think that needs a similar treatment, some are used by ifacestate I think